### ISSUE TITLE
Kicked out of PCC with error code 429

### ISSUE THREAD

**Issue by jorgenask86** [ugumba/homey-panasonic-comfort-cloud-alt#10]
Hi!
I have successfully installed this alternative PCC app on my Homy Pro 2023
I can successfully add my Panasonic heatpump to my Homey Pro and change temperature and other settings

But If I add the heatpump in a flow to controll my heatpump using external sensors arround the house, I soon get an error message 429.
Seems like Panasonic doesn't like the relatively frequent data traffic?
Also I see some error code 500 in the log

If I have logged into PCC on my phone I would also get told there was suspicious activity and that I am suspended from the account for a while (or something like that). 
Although this last time I was completely logged out of the phone app just too rule out that as an issue, even though the phone app and the homey app would be on different credentials/emails.

Anyhow I will include the log from the homey app and hope there could be a solution to this other than getting a sensibo to control the heatpump .

[log PCChomey app.txt](https://github.com/ugumba/homey-panasonic-comfort-cloud-alt/files/14324085/log.PCChomey.app.txt)

Kind regards 
Jørgen Ask

**Comment by Magnusri** [comment]
This is true.
It happens because the API is really not supposed to be used for anything but their official app...

Any calls that are considered different to how the app should behave proceeds to lock you out for around 24h. This includes some amount of calls within some time (I do not think anyone has identified the amount or the time span)

**Comment by Magnusri** [comment]
You might be able to automate this still if you make some variable to store your data in, and make sure to not send it to this app as often.

**Comment by ugumba** [comment]
I agree - it does sound like your flow is causing the app to call Panasonic's API too often, @jorgenask86.
I'll leave this issue open as a reminder to look at it.

If you can't control how often your flow is triggered, you can probably easily "debounce" the flow by starting a 30 second (or so) "Chronograph" timer at the very top.  Only start it if it is not already counting down.
Then trigger the PCC-Alt flow card when the timer reaches 0.

**Comment by jorgenask86** [comment]
I already tried with running the flow only every 2 and 3 minutes and it still kickes me out after a while.
I have basically given up and started using a sensibo to controll my heatpump, haha.

I could one day try to run it again with the alternative app and try to get the log from before I get kicked out :)
I will post here if I do.

Thanks!

**Comment by jarll** [comment]
I suspect the cause is what I reported and fixed in https://github.com/ugumba/homey-panasonic-comfort-cloud-alt/issues/37 in April 2025. If action cards were triggered too often (before the fix), you got endless number of fetch timers and thus massive traffic on the API which does't stop until the app is restarted. At that time I remember a lot of users got blocked from the Panasonic service because of this.

The patch in https://github.com/ugumba/homey-panasonic-comfort-cloud-alt/issues/37 (included since version 1.1.15) prevents this.

Have you tried installing newer version since yous last comment March 2024?

**Comment by jarll** [comment]
Assumed already fixed. Closing.
